src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,246 @@
+     * for Event task.
+     *
+     * @param dateTime String representation of DateTime
+     * @return Array of String with date and time separated.
+     * @throws DukeMissingDateTimeException If input does not contain time or date.
+     */
+    public static String[] eventDateTimeSplit(String dateTime) throws DukeMissingDateTimeException {
+        String[] splitDateTime = dateTime.split(" ", 2);
+        if (splitDateTime.length != 2) {
+            throw new DukeMissingDateTimeException();
+        }
+        String[] splitStartEnd = splitDateTime[1].split("-", 2);
+        if (splitStartEnd.length != 2) {
+            throw new DukeMissingDateTimeException();
+        }
+        return new String[]{splitDateTime[0], splitStartEnd[0], splitStartEnd[1]};
+    }
+
+    /**
+     * Interprets user input and returns command to be executed.
+     *
+     * @param input User input.
+     * @return Command to be executed.
+     * @throws DukeException If any exception occurs when interpreting user input.
+     */
+    public Command parse(String input) throws DukeException {
+        Instruction instruction = Instruction.valueOfLabel(input.split(" ")[0]);
+        switch (instruction) {
+        case LIST:
+            return new ListCommand();
+        case DONE:
+            int doneIndex = extractIndex(input, Instruction.DONE);
+            return new DoneCommand(doneIndex);
+        case HELP:
+            return new HelpCommand();
+        case TODO:
+            String toDoInfo = extractInfo(input, Instruction.TODO);
+            ToDo toDo = new ToDo(toDoInfo);
+            return new AddCommand(toDo);
+        case DEADLINE:
+            String[] deadlineInfo = extractDeadline(extractInfo(input, Instruction.DEADLINE));
+            String[] deadlineDateTime = deadlineDateTimeSplit(deadlineInfo[1]);
+            Deadline deadline = new Deadline(deadlineInfo[0], parseDate(deadlineDateTime[0]),
+                    parseTime(deadlineDateTime[1]));
+            return new AddCommand(deadline);
+        case EVENT:
+            String[] eventInfo = extractEvent(extractInfo(input, Instruction.EVENT));
+            String[] eventDateTime = eventDateTimeSplit(eventInfo[1]);
+            Event event = new Event(eventInfo[0], parseDate(eventDateTime[0]), parseTime(eventDateTime[1]),
+                    parseTime(eventDateTime[2]));
+            return new AddCommand(event);
+        case DELETE:
+            int deleteIndex = extractIndex(input, Instruction.DELETE);
+            return new DeleteCommand(deleteIndex);
+        case FILTER:
+            String filterDate = extractInfo(input, Instruction.FILTER);
+            return new FilterCommand(parseDate(filterDate));
+        case FIND:
+            return new FindCommand(extractInfo(input, Instruction.FIND));
+        case BYE:
+            return new ExitCommand();
+        default:
+            return new InvalidCommand();
```

Comment:
I believe it's better to add the ```//Fallthrough``` comment whenever there is a ```case``` statement without a ```break``` statement. 
It helps to make clear that this is intentional and not by accident :smile: 
Example:
```javascript
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
```
Saw this from the [textbook](https://se-education.org/guides/conventions/java/intermediate.html#layout).